logger.js

```diff
@@ -1,111 +1,262 @@
 'use strict';
 
-const winston = require('winston');
+const { createLogger, format, transports } = require('winston');
 
 function convertToBoolean(value) {
-  return typeof value === 'string' ? value === 'true' : value;
+    return typeof value === 'string' ? value === 'true' : value;
 }
 
 function validateHttpConfig(httpOptions) {
-  if(!httpOptions.hasOwnProperty('host')) {
-    throw new Error('The property host is omitted from config option log2httpServer. Please fill this field or remove the property.');
-  }
-}
-
-function generateConsoleTransport(service) {
-  return new winston.transports.Console({
-    timestamp: function () {
-      return new Date().toISOString();
-    },
-    json: false,
-    formatter: function (options) {
-      let logMessage = `[${options.timestamp()}] [${options.level}] [${service.name}] [${service.version}] ${options.message}`;
-      if (options.meta && Object.keys(options.meta).length) {
-        logMessage += ` [${JSON.stringify(options.meta)}]`;
-      }
-      return logMessage;
-    }
-  });
+    if (!httpOptions.hasOwnProperty('host')) {
+        throw new Error('The property host is omitted from config option log2httpServer. Please fill this field or remove the property.');
+    }
 }
 
 function generateFileTransport(level) {
-  const path = require('path');
-  return new winston.transports.File({
-    name: `${level}-file`,
-    filename: path.resolve(`./logs/filelog-${level}.log`),
-    maxsize: 5242880, // 5MB
-    maxFiles: 5,
-    level: level
-  });
+    const path = require('path');
+    return new transports.File({
+        filename: `filelog-${level}.log`,
+        dirname: path.resolve(`./logs`),
+        maxsize: 5242880, // 5MB
+        tailable: true
+    });
 }
 
 function generateHttpTransport(serverLogConfig) {
-  return new winston.transports.Http(serverLogConfig);
+    new transports.http(serverLogConfig);
 }
 
-module.exports.MCLogger = class MCLogger extends winston.Logger {
-  constructor(config, service) {
+module.exports.MCLogger = class MCLogger {
+
+    constructor(config, service) {
+        if (!service) {
+            throw new Error('Config is required.');
+        }
+        if (!config) {
+            throw new Error('Config is required.');
+        }
+        if (!config.level) {
+            throw new Error('Level property is required in config.');
+        }
+
+        const params = {
+            level: config.level || 'debug',
```

Comment:
default should be error or warn as it is the main configuration that will be used on prod